prerendering-state.md

```diff
@@ -0,0 +1,150 @@
+
+This [background
+doc](https://docs.google.com/document/d/1Xzw0k8DgltI2ohapuDKmjRZLv7NVrRFGusW8IBtiCT0/edit#heading=h.acmnp6zdmcik) deals
+extensively with this issue. An ideal solution would introduce a new event for this use case, rather than implicitly
+tying it to any particular state.
+
+### Relationship to loading-modes
+
+[Alternate-loading-modes](https://github.com/bokand/alternate-loading-modes/blob/main/browsing-context.md#current-proposal)
+proposes explicit APIs to expose the exact loading mode used. Even with `document.prerendering`, this additional
+information may still be useful for specialized cases where fetching may need to depend on the exact loading mode (TODO:
+examples?).
+
+However, we believe `document.prerendering` will cover the most common use cases without getting into the level of
+granularity of fetching restrictions.
+
+Note: it's possible that `document.prerendering` may be  just syntactic sugar for some subset of
+`document.loadingMode.type`, e.g. `document.loadingMode.type != 'default'` or `document.loadingMode.type ==
+'prerender' || document.loadingMode.type == 'uncredentialed-prerender'`.
+
+### Relationship to other states
+
+As noted in the [background
+doc](https://docs.google.com/document/d/1Xzw0k8DgltI2ohapuDKmjRZLv7NVrRFGusW8IBtiCT0/edit#heading=h.14z99pd6akf0), there
+are other states where it may make sense to report `prerendering` state. For example, a page loading into a mobile
+background tab maybe terminated, without notice, before the user ever sees it. Similarly, the live-preview in a mobile app
+or tab switcher is a non-interactive preview, reminiscent of a portal.
+
+Should these states also report `prerendering==true`? Should they use the prerendering browsing context concept? The
+answers here aren't clear but this is worth considering.
+
+### Naming bikeshed
+
+We've called this state `prerendering` here to make clear the association between it and a [prerendering browsing
+context](browsing-context.md). However, this _may_ be misleading in some cases. Portals aren't exactly prerendering in the way most authors
+would think about it. Additionally, a portal may enter the prerendering state from a regular context; this feels
+unintuitive to call "prerendering".
+
+We should also consider future use cases. For example, if
+[fenced-frames](https://github.com/shivanigithub/fenced-frame/) were to use this special browsing context, does it make
+sense to call it "prerendering"?
+
+The term "interactive" seems appealing as it captures a major difference between prerendering/portals and regular
+browsing contexts. However, it would be unintuitive to explain how a background tab is considered "interactive". In
+particular, if we want this to solve [PageVisibility#59](https://github.com/w3c/page-visibility/issues/59) we'd have the
+bizarre case that entering the app switcher would make a page non-interactive, then backgrounding it from there would make
+it interactive.
```

Comment:
It might be worth mentioning how it does (or doesn't, IIUC) interact with bfcache.

Reply:
I've been thinking of `document.prerendering` as "is in prerendering browsing context". Does BFCache have any expected interaction with the concept of a prerendering browsing context introduced in this repo? I expect not (and in that case I'll add that here) but I'm not sure if we're thinking of reusing some of the rendering/permissions restrictions.

Reply:
I think likely not. Though it's similar in some regards to prerendering, script can't observe that much because it gets stopped for the duration where that would apply. But I think it's non-obvious enough to note.

Reply:
+1 could be good to mention this. My understanding also is that bfcached pages do not run script so can't observe document.prerendering.